nltk/translate/bleu_score.py

```diff
@@ -379,3 +398,132 @@ def _brevity_penalty(closest_ref_len, hyp_len):
     else:
         return math.exp(1 - closest_ref_len / hyp_len)
 
+
+def smooth_precision(references, hypothesis, p_n, hyp_len, 
+                    method=0, epsilon=0.1, alpha=5, k=5):
+    """
+    This is an implementation of the smoothing techniques 
+    for segment-level BLEU scores that was presented in 
+    Boxing Chen and Collin Cherry (2014) A Systematic Comparison of 
+    Smoothing Techniques for Sentence-Level BLEU. In WMT14. 
+    http://acl2014.org/acl2014/W14-33/pdf/W14-3346.pdf
+
+        >>> hypothesis1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'which',
+        ... 'ensures', 'that', 'the', 'military', 'always',
+        ... 'obeys', 'the', 'commands', 'of', 'the', 'party']
+        >>> reference1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'that',
+        ... 'ensures', 'that', 'the', 'military', 'will', 'forever',
+        ... 'heed', 'Party', 'commands']
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=0)) # doctest: +ELLIPSIS
+        0.4118...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=1)) # doctest: +ELLIPSIS
+        0.4118...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=2)) # doctest: +ELLIPSIS
+        0.4576...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=3)) # doctest: +ELLIPSIS
+        0.4118...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=4)) # doctest: +ELLIPSIS
+        0.4118...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=5)) # doctest: +ELLIPSIS
+        0.4905...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=6)) # doctest: +ELLIPSIS
+        0.1801...
+        >>> print (sentence_bleu([reference1], hypothesis1, smoothing_method=7)) # doctest: +ELLIPSIS
+        0.4905...
+    
+    :param references: reference sentences
+    :type references: list(list(str))
+    :param hypothesis: a hypothesis sentence
+    :type hypothesis: list(str)
+    :param p_n: the list of modified precision
+    :type p_n: Fraction
+    :param hyp_len: the length of the hypothesis
+    :type hyp_len: int
+    :param method: the smoothing method to be used
+    :type method: int
+    :param epsilon: the epsilon value use in method 1
+    :type epsilon: float
+    :param alpha: the alpha value use in method 6
+    :type alpha: int
+    :param k: the k value use in method 4
+    :type k: int
+    :return: a list of smoothed modified precisions
+    :rtype: list(float)
+    """
+    # No smoothing.
+    if method == 0:
```

Comment:
Actually the numbers should be preferred over boolean labels because the default values are set empirically from experiments. So i would expect users to try out multiple values instead of using the values set from Chen and Cherry.